api/agent_client.go

```diff
@@ -422,6 +425,28 @@ func (c *AgentClient) FailJob(ctx context.Context, jobUUID string, errorDetail s
 	return err
 }
 
+// CreateStackNotification creates a stack notification for a job.
+// This is designed to be call and forget, error will be logged but we don't anticipate error handling
+func (c *AgentClient) CreateStackNotification(ctx context.Context, jobUUID string, detail string) {
+	if !c.UseStackAPI() {
+		// No-op for the case for legacy API
+		return
+	}
+
+	go func() {
```

Comment:
Why does this need to be concurrent?

Reply:
Since this is method is a auxiliary thing, I don't want caller to be blocked by this.

Reply:
Why _shouldn't_ the caller be blocked? All that's going to do is slow down processing of jobs. If notifications are taking a long time, then maybe we shouldn't hurry the rest of the loop.

If we do decide to not block the caller for whatever reason, then there should be a fixed size pool of goroutines for sending notifications, not one goroutine per notification, since that risks filling the runtime with goroutines.

Reply:
If you feel strongly about sending the notifications concurrently to the rest of the loop, I'm happy to compromise on the "why".

I'm less compromising on the "how". If there's no need to send more than a few notifications at the same time, then there's no need to create a potentially unbounded number of goroutines to do that. What I mean is, this implementation is not creating 1 extra goroutine per _job_, it's creating 1 extra goroutine per `CreateStackNotification` call. Any bug that causes `CreateStackNotification` to be called many times repeatedly could blow up the runtime.

It may not be a problem for agent-stack-k8s if we only call this once per job, but this will certainly be treated as example code by other implementers, so I would like to do it right.

Reply:
Fair point, I've made a goroutine pool for this. 

Although, as I am making it, I realized that we should probably make that endpoint batched too.